akka-persistence/src/main/scala/akka/persistence/snapshot/local/LocalSnapshotStore.scala

```diff
@@ -126,9 +127,13 @@ private[persistence] class LocalSnapshotStore extends SnapshotStore with ActorLo
   private def snapshotMetadatas(persistenceId: String, criteria: SnapshotSelectionCriteria): immutable.Seq[SnapshotMetadata] = {
     val files = snapshotDir.listFiles(new SnapshotFilenameFilter(persistenceId))
     if (files eq null) Nil // if the dir was removed
-    else files.map(_.getName).collect {
-      case FilenamePattern(pid, snr, tms) ⇒ SnapshotMetadata(URLDecoder.decode(pid, UTF_8), snr.toLong, tms.toLong)
-    }.filter(md ⇒ criteria.matches(md) && !saving.contains(md)).toVector
+    else {
+      files.map(_.getName).flatMap { filename ⇒
```

Comment:
thinking that maybe using `.iterator` would save a bit of collection allocations, so that only the final `.toVector` creates a collection. Not sure it is relevant performance wise.